### PR TITLE
294 update bundles to use BundleWorkflow interface

### LINKED ISSUE #294: Define `initialize`, `run`, `finalize` sections in bundle config
**Is your feature request related to a problem? Please describe.**
Currently, to support code execution in order, we defined expressions in a list of bundle config, like:
https://github.com/Project-MONAI/model-zoo/blob/dev/models/spleen_ct_segmentation/configs/multi_gpu_train.json#L27
But when loading the bundle in other applications, like MONAI Label, MONAI FL, we want to get the `trainer` or other components and hard to execute all the expressions in the list immediately, so we re-define much logic in the python parser, like:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/fl/client/monai_algo.py#L460

To easily connect with other apps, I think we should clearly define the expressions in `initialize`, `run`, `finalize` sections, then others can call the operations separately. For example:
```json
{
    "initialize": [
        "$import torch.distributed as dist",
        "$dist.is_initialized() or dist.init_process_group(backend='nccl')",
        "$torch.cuda.set_device(@device)",
        "$monai.utils.set_determinism(seed=123)",
        "$setattr(torch.backends.cudnn, 'benchmark', True)",
        "$import logging",
        "$@train#trainer.logger.setLevel(logging.WARNING if dist.get_rank() > 0 else logging.INFO)",
        "$@validate#evaluator.logger.setLevel(logging.WARNING if dist.get_rank() > 0 else logging.INFO)"
    ],
    "run": [
        "$@train#trainer.run()"
    ],
    "finalize": [
        "$dist.destroy_process_group()"
    ]
}
```
Will also create another ticket to define a high level API `Bundle` to simplify the bundle loading in other apps.

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Part of #294 .

### Description
This PR updates several bundles with `BundleWorkflow ` interface, and also fix a few errors. Commands in all bundles are tested with `monai==1.2.0rc3` locally with the changes except the following two:
~~1. `wholeBody_ct_segmentation`~~ (checked)
~~2. `pathology_nuclei_segmentation_classification`~~ (checked)
~~because I do not have these two datasets locally.~~

### Status
**Ready**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: models/swin_unetr_btcv_segmentation/configs/metadata.json [hunk 1]
```diff
@@ -19,12 +20,12 @@
...
+    "pytorch_version": "1.13.1",
+    "numpy_version": "1.22.4",
     "optional_packages_version": {
-        "nibabel": "3.2.1",
-        "pytorch-ignite": "0.4.8",
+        "nibabel": "4.0.1",
+        "pytorch-ignite": "0.4.9",
         "einops": "0.4.1"
     },
     "name": "Swin UNETR BTCV segmentation",
```
Comment (yiheng-wang-nv) [inline models/swin_unetr_btcv_segmentation/configs/metadata.json:32]: Hi @tangy5 , as for #340 , if anything need to be updated, feel free to push a commit into this PR directly, thanks!

Comment (tangy5) [reply models/swin_unetr_btcv_segmentation/configs/metadata.json:32]: @yiheng-wang-nv , I've updated the numbers in this PR. Thanks.

File: models/brats_mri_segmentation/configs/multi_gpu_train.json [hunk 2]
```diff
@@ -24,13 +24,17 @@
     },
     "validate#dataloader#sampler": "@validate#sampler",
     "validate#evaluator#val_handlers": "$None if dist.get_rank() > 0 else @validate#handlers",
-    "training": [
+    "initialize": [
         "$import torch.distributed as dist",
         "$dist.init_process_group(backend='nccl')",
```
Comment (Nic-Ma) [inline models/brats_mri_segmentation/configs/multi_gpu_train.json]: Please update this referring to the spleen bundle:
https://github.com/Project-MONAI/model-zoo/blob/dev/models/spleen_ct_segmentation/configs/multi_gpu_train.json#L29

Thanks.

File: models/endoscopic_inbody_classification/configs/multi_gpu_train.json [hunk 3]
```diff
@@ -24,13 +24,17 @@
     },
     "validate#dataloader#sampler": "@validate#sampler",
     "validate#evaluator#val_handlers": "$None if dist.get_rank() > 0 else @validate#handlers",
-    "training": [
+    "initialize": [
         "$import torch.distributed as dist",
         "$dist.init_process_group(backend='nccl')",
```
Comment (Nic-Ma) [inline models/endoscopic_inbody_classification/configs/multi_gpu_train.json]: Same comment for all the bundles.

Thanks.

**Review by Nic-Ma — COMMENTED** [review]
Hi @yiheng-wang-nv ,

Thanks for the big effort, please update the "$dist.init_process_group(backend='nccl')" line for every multi-gpu training or multi-gpu evaluation for every bundle.
Others look good to me.

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update, looks good to me.
But I would suggest to wait for the MONAI 1.2rc3 tonight @wyli .
Then update every bundle for the MONAI version and PyTorch version in this PR.
Better to verify several typical bundles again with rc3 docker.

Thanks.

**Comment by yiheng-wang-nv** [comment]
Hi @tangy5 , this PR will modify several bundle's format. Could you please help to check if MONAI label is compatible with both new and old formats? Thanks!

**Comment by tangy5** [comment]
HI @yiheng-wang-nv , thanks. Users of MONAI Label can download old version bundles, by adding a version tag, no change for that. I will create a PR to change the bundle training config for a bit, it should be all good.

**Review by tangy5 — APPROVED** [review]

**Comment by yiheng-wang-nv** [comment]
Hi @drbeh , I check the tumor detection bundle and met an error when running the script: `evaluate_froc.sh`

I checked the code and think that this place is wrong:
https://github.com/Project-MONAI/model-zoo/blob/b75873cb677af6f26508517535fab906b83b0d55/models/pathology_tumor_detection/scripts/lesion_froc.py#L20

There is no suffix for "tumor_mask", and I checked the source code of `LesionFROC` and it does not supply the suffix (https://github.com/Project-MONAI/MONAI/blob/05533abc2974631cf65d08d8719c09a27f16bed9/monai/apps/pathology/metrics/lesion_froc.py#L110) Therefore, error will be raise as follow:
```
ValueError: Cannot find a plugin to handle '/workspace/data/testing/ground_truths/test_001'!
```
Therefore, I made a change in:
https://github.com/Project-MONAI/model-zoo/pull/339/files#diff-f5293514d237b45495514ff3696114893dbe4740564e6712121bc39adbfadb5bR20 and successfully run the script. Could you please help to double confirm if the change is correct? Thanks!

**Comment by bhashemian** [comment]
> Hi @drbeh , I check the tumor detection bundle and met an error when running the script: `evaluate_froc.sh`
> 
> I checked the code and think that this place is wrong:
> 
> https://github.com/Project-MONAI/model-zoo/blob/b75873cb677af6f26508517535fab906b83b0d55/models/pathology_tumor_detection/scripts/lesion_froc.py#L20
> 
> There is no suffix for "tumor_mask", and I checked the source code of `LesionFROC` and it does not supply the suffix (https://github.com/Project-MONAI/MONAI/blob/05533abc2974631cf65d08d8719c09a27f16bed9/monai/apps/pathology/metrics/lesion_froc.py#L110) Therefore, error will be raise as follow:
> 
> ```
> ValueError: Cannot find a plugin to handle '/workspace/data/testing/ground_truths/test_001'!
> ```
> 
> Therefore, I made a change in: https://github.com/Project-MONAI/model-zoo/pull/339/files#diff-f5293514d237b45495514ff3696114893dbe4740564e6712121bc39adbfadb5bR20 and successfully run the script. Could you please help to double confirm if the change is correct? Thanks!

Hi @yiheng-wang-nv, you are right, the reason for this is behavior is that the name that is used in ProbMapProducer used to be `xxx.tif.npy` but it has changed to `xxx.npy` so your change is correct. Thanks for fixing it.

**Review by Nic-Ma — APPROVED** [review]

**Comment by yiheng-wang-nv** [comment]
Hi @YanxuanLiu the latest blossom ci failed on oss vulnerability scan, could you please help to check the error? https://github.com/Project-MONAI/model-zoo/actions/runs/4597307449/jobs/8119836797#step:4:345 Thanks in advance!

**Comment by yiheng-wang-nv** [comment]
> Hi @YanxuanLiu the latest blossom ci failed on oss vulnerability scan, could you please help to check the error? https://github.com/Project-MONAI/model-zoo/actions/runs/4597307449/jobs/8119836797#step:4:345 Thanks in advance!

It seems the issue is from the blossom side, not related to the content of this PR. I will merge this PR first. cc @Nic-Ma @YanxuanLiu @pxLi

**Pull request merged**